### ISSUE TITLE
Show button doesn't work

### ISSUE THREAD

**Issue by roguedevjake** [spetzel2020/quick-encounters#108]
**Foundry:** 10.285
**System:** D&D 5e 2.0.2
**Modules:** Quick Encounters 1.0.5 (only)

The show button doesn't seem to work no matter what, and if hide is pressed the encounter is no longer accessible at all.

Cancel is also problematic if automatically opening encounters is not selected, as it will do a similar thing.

**Comment by spetzel2020** [comment]
Thanks for the clear report - will take a look.

**Comment by spetzel2020** [comment]
Seems to be related to the following:
```
    static async getJournalSheetHeaderButtons(journalSheet, buttons) {
        //0.7.3: Add a Show QE button if this JE has a Quick Encounter and showQEAutomatically is false OR the QE has been hidden
        const quickEncounter = QuickEncounter.extractQuickEncounter(journalSheet);
        const displayShowQEButton = !game.settings.get(QE.MODULE_NAME,"showQEAutomatically") || quickEncounter?.hideQE;
```
 The above adds a Show button on the Journal Entry frame if either the Setting Show QE is false OR the current Quick Encounter is hidden. However, with QEs now being associated with Journal Entry Pages, the current Quick Encounter could be shown (so QE wouldn't show the Show button), but then if you switch to a different Journal Entry Page **that** QE might be hidden but the Show button still won't be there.

Solutions:

- Simplest would be to always add the Show button
- Trigger the show/hide of the Show button when you display the QE

**Comment by spetzel2020** [comment]
The second problem is that when the Show button is clicked, it tries to retrieve the Journal Sheet and will get the Journal Entry Sheet, not the sub-sheet we're currently in.

Also we could be looking at the Journal Entry in "flow mode" where all Journal Sheets (and all QEs) are showing. In this case, Show should re-open all the missing QEs.

Unfortunately in Foundry v10 I think we need to remove the Hide button (you can get almost the same behavior by pressing Cancel to always close the QE). And Show will open the QEs for all active JE Pages.

**Comment by spetzel2020** [comment]
IN QE 1.1.0 and Foundry v10, Show now shows all QEs associated with open Journal Entry Pages (including all of them when in Journal Entry flow mode)

If you clear the Show QE Automatically setting, then the Show button is always visible in the JE border.

**Comment by spetzel2020** [comment]
Fixed in 1.1.0b (not released)
